src/items/mod.rs

```diff
@@ -345,11 +232,37 @@ fn last_day_of_month(year: i32, month: u32) -> u32 {
         .day()
 }
 
-fn at_date_inner(date: Vec<Item>, mut d: DateTime<FixedOffset>) -> Option<DateTime<FixedOffset>> {
-    d = d.with_hour(0).unwrap();
-    d = d.with_minute(0).unwrap();
-    d = d.with_second(0).unwrap();
-    d = d.with_nanosecond(0).unwrap();
+fn at_date_inner(date: Vec<Item>, at: DateTime<FixedOffset>) -> Option<DateTime<FixedOffset>> {
+    let mut d = at
+        .with_hour(0)
+        .unwrap()
+        .with_minute(0)
+        .unwrap()
+        .with_second(0)
+        .unwrap()
+        .with_nanosecond(0)
+        .unwrap();
+
+    // This flag is used by relative items to determine which date/time to use.
+    // If any date/time item is set, it will use that; otherwise, it will use
+    // the `at` value.
+    //
+    // TODO: find cleaner way to do this
+    let mut date_time_set = false;
+    for item in &date {
+        match item {
+            Item::Timestamp(_)
+            | Item::Date(_)
+            | Item::DateTime(_)
+            | Item::Year(_)
+            | Item::Time(_)
+            | Item::Weekday(_) => {
+                date_time_set = true;
+                break;
+            }
+            _ => {}
+        }
+    }
```

Comment:
A cleaner way might be an iterator with `any`:
```suggestion
    let date_time_set = date.iter().any(|item| {
        matches!(
            item,
            Item::Timestamp(_)
                | Item::Date(_)
                | Item::DateTime(_)
                | Item::Year(_)
                | Item::Time(_)
                | Item::Weekday(_)
        )
    });
```

Reply:
:+1: